File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = {};
        if(denom1==denom2){
            answer[0]=numer1+numer2;
            answer[1]=denom1;
        }
        else{
            int i= denom1*denom2;
            int j= (numer1*denom2)+(numer2*denom1);
            for(int k=1; k<j; k++){
                if(j%k==0){
                    j=j/k;
                    i=i/k;
                }
                    
                    
            }           
            
            answer[0]=j;
            answer[1]=i;
            }
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    answer = []
    denom=denom1*denom2
    numer=numer1*denom2+numer2*denom1
    
    for i in range(denom,0,-1):
        if (denom%i==0)and(numer%i==0):
            denom=denom/i
            numer=numer/i
        else:
            continue
    answer=[numer,denom]
            
    return answer